### PR TITLE
Catch exception on login when keyring fails to get a valid storage

### LINKED ISSUE #1197: Login not possible when keyring cannot find a valid storage
When login on a server, I get this exception:
```
In [1]: from astroquery.eso import Eso
eso                                                                        
In [2]: eso = Eso()                                                                                         
                         
In [3]: eso.login(username='MUSE-XXX')                                        
---------------------------------------------------------------------------
DBusErrorResponse                         Traceback (most recent call last)
~/miniconda3/lib/python3.6/site-packages/secretstorage/util.py in send_and_get_reply(self, msg)                    
     40                 try:                                           
---> 41                         return self._connection.send_and_get_reply(msg)
     42                 except DBusErrorResponse as resp:

~/miniconda3/lib/python3.6/site-packages/jeepney/integrate/blocking.py in send_and_get_reply(self, message)                 
     48                      
---> 49         return future.result()                                  
     50                                            

DBusErrorResponse: [org.freedesktop.DBus.Error.UnknownMethod] ("No such interface 'org.freedesktop.DBus.Properties' on object at path /org/freedesktop/secrets/collection/login",)

The above exception was the direct cause of the following exception:
...

InitError                                 Traceback (most recent call last)
<ipython-input-3-9186902b997a> in <module>()
----> 1 eso.login(username='MUSE-GTO')

~/lib/astroquery/astroquery/query.py in login(self, *args, **kwargs)
    335     def login(self, *args, **kwargs):
    336         with suspend_cache(self):
--> 337             self._authenticated = self._login(*args, **kwargs)
    338         return self._authenticated
    339

~/lib/astroquery/astroquery/eso/core.py in _login(self, username, store_password, reenter_password)
    220         if reenter_password is False:
    221             password_from_keyring = keyring.get_password(
--> 222                 "astroquery:www.eso.org", username)
    223         else:
    224             password_from_keyring = None

~/miniconda3/lib/python3.6/site-packages/keyring/core.py in get_password(service_name, username)
     39     """Get password from the specified service.
     40     """
---> 41     return _keyring_backend.get_password(service_name, username)
     42
     43

~/miniconda3/lib/python3.6/site-packages/keyring/backends/SecretService.py in get_password(self, service, username)
     65         """Get password of the username for the service
     66         """
---> 67         collection = self.get_preferred_collection()
     68         items = collection.search_items(
     69             {"username": username, "service": service})

~/miniconda3/lib/python3.6/site-packages/keyring/backends/SecretService.py in get_preferred_collection(self)
     55                 collection = secretstorage.get_default_collection(bus)
     56         except exceptions.SecretStorageException as e:
---> 57             raise InitError("Failed to create the collection: %s." % e)
     58         if collection.is_locked():
     59             collection.unlock()

InitError: Failed to create the collection: Prompt dismissed..
```

I don't know why exactly (no dbus, gnome-keyring or whatever) but this should not prevent to login without storing the password. I can workaround this by using `reenter_password=False` but this is not obvious (login parameters are not documented).
So I guess the simple solution would be to try/except the `keyring.get_password` call :
```
        # Get password from keyring or prompt
        if reenter_password is False:
            password_from_keyring = keyring.get_password(
                "astroquery:www.eso.org", username)
        else:
            password_from_keyring = None
```

### PR DESCRIPTION
**Pull request by saimn**
Fix #1197.

### CODE REVIEW

File: astroquery/eso/core.py [hunk 1]
```diff
@@ -216,11 +216,13 @@ def _login(self, username=None, store_password=False,
...
         if reenter_password is False:
-            password_from_keyring = keyring.get_password(
-                "astroquery:www.eso.org", username)
-        else:
-            password_from_keyring = None
+            try:
+                password_from_keyring = keyring.get_password(
+                    "astroquery:www.eso.org", username)
+            except keyring.errors.KeyringError:
+                log.info("Failed to get a valid keyring for password storage")
```
[the remark this answers is not in the document]
Comment (saimn) [reply astroquery/eso/core.py]: Ok, done.

**Comment by bsipocz** [comment]
@saimn @keflavich - Is this only an issue for the `eso` module, or everything that uses keyring?

If the latter, would you be interested in fixing up the rest, too? Quick search says they are: `alma`, `cosmosim`, `mast`, `nrao`

**Comment by saimn** [comment]
@bsipocz - I had this idea too, but the code is not factorized, so I refrained myself to look at the other modules ;). I guess this could go in a helper method in `QueryWithLogin`, maybe for a follow-up PR ?

**Comment by bsipocz** [comment]
@saimn - A follow-up PR sounds perfectly reasonable to me.

**Comment by bsipocz** [comment]
Thanks @saimn!

**Pull request merged**